### PR TITLE
In the Feast google update subs lambda, get the V1 subscription for historical queue

### PR DESCRIPTION
**Pull request by MaelGNM**
## What does this change?
In https://github.com/guardian/mobile-purchases/pull/1654 we started sending Feast subscription to the "historical" queue for them to be recorded in GCP. However for the Google version we use a different version of the API from the live app subscriptions, meaning the data is in a different format and not parsed correctly. Resulting in missing data in the lake.

In this PR we also fetch the V1 subscription specifically to be sent to the "historical" queue.

Longer term we should consider migrating all the lambdas to use the V2 google APIs.

## How can we measure success?

Data for Feast subscriptions is visible in BigQuery.

### CODE REVIEW

File: typescript/src/feast/update-subs/google.ts [hunk 1]
```diff
@@ -33,8 +34,48 @@ const googleSubscriptionToSubscription = (
     )
 };
 
+const googleResponseBodyToSubscription = (
```
Comment (tjmw) [inline typescript/src/feast/update-subs/google.ts:37]: Maybe not for this PR, but would it be worth a follow up to refactor away the duplication between this file and `src/update-subs/google.ts`? It looks like this response parsing code exists in both.

**Review by tjmw — APPROVED** [review]
Great 👍

File: typescript/src/feast/update-subs/google.ts (hunk 1 above)
Comment (MaelGNM) [reply typescript/src/feast/update-subs/google.ts:37]: Yes, I had considered it but got a bit lost in the code so this way was easier. I'll have a go at refactoring.

Comment (tjmw) [reply typescript/src/feast/update-subs/google.ts:37]: 👍 Yeah totally understood! As I said I think fine to get this out and attempt the refactor in a follow up PR.

**Review by tjmw — APPROVED** [review]

**Pull request merged**